### PR TITLE
[@types/node] Add missing parameters for process.emitWarning

### PR DESCRIPTION
**Pull request by murgatroid99**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/process.html#process_process_emitwarning_warning_options, https://nodejs.org/api/process.html#process_process_emitwarning_warning_type_code_ctor
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by SimonSchick — CHANGES_REQUESTED** [review]
Test pls

File: types/node/globals.d.ts [hunk 1]
```diff
@@ -841,6 +841,13 @@ declare namespace NodeJS {
         voluntaryContextSwitches: number;
     }
 
+    interface ProcessWarningOptions {
+        type?: String;
```
Comment (SimonSchick) [inline types/node/globals.d.ts]: This should be `string` not `String`.

**Review by SimonSchick — CHANGES_REQUESTED** [review]

**Comment by murgatroid99** [comment]
Type fixed and some tests added

**Comment by KSXGitHub** [comment]
From the docs:

> If warning is passed as an `Error` object, it will be passed through to the `'warning'` event handler unmodified (and the optional `type`, `code` and `ctor` arguments will be ignored)

I think you should remove `Error` from `string | Error` and add an overload for it:

```typescript
emitWarning(warning: Error): void;
```

**Review by SimonSchick — APPROVED** [review]

**Comment by murgatroid99** [comment]
I disagree with separating out the `Error` override like that. It is still valid in either case to pass the extra arguments if the first argument is an `Error`.

**Pull request closed without merging**